WordPress/gutenberg
# Fix error when block is an email input

## Description


Closes #26074
Closes #27125

According to https://html.spec.whatwg.org/multipage/input.html#input-type-attr-summary, an input element with `type="email"` doesn't support selection. The description of the `isTextField` function states that:

> Check whether the given element is a text field, **where text field is defined by the ability to select within the input, or that it is contenteditable.**

Since the `email` type doesn't pass that, I'm changing it there. So, when an email input is used as the first tabbable element in a block, instead of treating it as normal text input, it behaves just like other non-text inputs (like checkbox, radio etc.), which means the user has to press Tab to reach it.

## How has this been tested?




Same instructions as #26074:

1. Create a new post
2. Copy this into the console:

    ```js
    wp.blocks.registerBlockType(
    	'example/email-input-reproduce-issue',
    	{
    		title: 'Email Input',
    		category: 'common',
    		icon: 'wordpress-alt',
    		keywords: [ 'Foo' ],
    		attributes: {
    			'email': {
    				type: 'string',
    			},
    		},
    		edit: function( props ) {
    			return wp.element.createElement( 'input', { type: 'email', value: props.email } );
    		},
    	}
    );
    ```

3. Insert the 'Email Input' block you just registered:

    <img width="554" alt="Screenshot of the editor with the quick inserter open focusing on the newly created Email Input block" src="https://user-images.githubusercontent.com/4063887/95902116-f181d400-0d59-11eb-91b0-61af9a1c6a7d.png">

4. Expected: The block should display
5. Actual: There's a block and console error:

    <img width="818" alt="Screenshot of the editor showing the error on the editor interface and on the browser console" src="https://user-images.githubusercontent.com/4063887/95902340-4de4f380-0d5a-11eb-9734-1e1afca82e98.png">

## Screenshots

![Moving through 2 paragraph blocks with a custom email input block in between using arrow keys and tab](https://user-images.githubusercontent.com/3068563/106279173-c80e8a80-621a-11eb-94d7-e4ecc2426ff8.gif)

## Types of changes





Bug fix

## Checklist:
